### DISCUSSION TITLE
Xamarin / C# / .Net version compatibility?

### DISCUSSION THREAD

**Discussion by vteague** [microsoft/electionguard#129] [Q&A] [answered]
Dear ElectionGuard people, I have a bit of a newbies' question about which version of .Net I should be using.  As you know, I'm writing an app in Xamarin and intending to build for Android, iOS and UWP.

Right now, I'm on netstandard2.0 and everything is working fine (at least on iOS and Android).

However, I'd like to up my C# version to at least 8.0 (mostly so I can enable Nullable). My question is, will this break anything when I incorporate the ElectionGuard library?

Please advise. Don't be embarrassed to give me insultingly specific instructions...

**Comment by keithrfung** [comment, marked as answer]
@vteague So netstandard2.0 can support many different targets. Here is the table:
https://github.com/dotnet/standard/blob/master/docs/versions.md

You can specifically target a version of .Net to the language you'd like. I would actually suggest going for .NET 5.0. 
https://en.wikipedia.org/wiki/C_Sharp_(programming_language)

So instead of targeting netstandard2.0, you can target net50
